pkg/github/client/rate_limit_watcher_client.go

```diff
@@ -16,24 +16,37 @@ type rateLimitWatcher struct {
 
 // NewRateLimitWatcherClient wraps github client with calls watching GH API rate limits
 func NewRateLimitWatcherClient(c Client, log log.Logger, threshold int) Client {
-	return &rateLimitWatcher{Client: c, log: log, threshold: threshold}
+	return rateLimitWatcher{Client: c, log: log, threshold: threshold}
 }
 
 func (r rateLimitWatcher) logRateLimitsAfter(f func()) {
 	f()
 	r.logRateLimits()
 }
 
-func (r rateLimitWatcher) logRateLimits() {
+// GetRateLimits retrieves the rate limits for the current GH client
+func GetRateLimits(c Client) *gogh.RateLimits {
```

Comment:
I'm slightly confused here - what was wrong with the previous solution https://github.com/arquillian/ike-prow-plugins/pull/142/commits/c499fdf0e20309722e9154e5304bdd61222e0305#diff-535bf210f0256dc391d9eff7602869f5L36